src/main/java/org/terasology/alterationEffects/boost/HealthBoostAlterationEffect.java

```diff
@@ -1,160 +1,77 @@
+// SPDX-License-Identifier: Apache-2.0
 package org.terasology.alterationEffects.boost;
 
 import org.terasology.alterationEffects.AlterationEffect;
 import org.terasology.alterationEffects.AlterationEffects;
+import org.terasology.alterationEffects.ComponentBasedAlterationEffect;
+import org.terasology.alterationEffects.EffectContext;
 import org.terasology.alterationEffects.OnEffectModifyEvent;
 import org.terasology.context.Context;
 import org.terasology.engine.Time;
 import org.terasology.entitySystem.entity.EntityRef;
-import org.terasology.logic.delay.DelayManager;
 import org.terasology.logic.health.HealthComponent;
 import org.terasology.math.TeraMath;
 
+import java.util.Optional;
+
 /**
  * This handles the application of the health boost effect, which boosts an entity's maximum health (based on the
  * magnitude) for a specified duration.
  */
-public class HealthBoostAlterationEffect implements AlterationEffect {
+public class HealthBoostAlterationEffect extends ComponentBasedAlterationEffect<HealthBoostComponent> implements AlterationEffect {
 
     private final Time time;
-    private final DelayManager delayManager;
 
     /**
      * Constructor. Instantiate an instance of this alteration effect using the provided context. This context will be
      * used to get the time and DelayManager.
      *
-     * @param context       The context which this effect will be executed on.
+     * @param context The context which this effect will be executed on.
      */
     public HealthBoostAlterationEffect(Context context) {
+        super(context, HealthBoostComponent.class, AlterationEffects.MAX_HEALTH_BOOST);
         this.time = context.get(Time.class);
-        this.delayManager = context.get(DelayManager.class);
     }
 
     /**
      * Removes the effect of the given health boost from the entity.
      *
-     * @param entity        The entity which has the health boost effect.
-     * @param hBoost        The health boost component.
+     * @param entity The entity which has the health boost effect.
+     * @param boost The health boost component.
      */
-    private void removeBoost(EntityRef entity, HealthBoostComponent hBoost) {
-        HealthComponent h = entity.getComponent(HealthComponent.class);
-
-        // Reverse the max health boosting effect by dividing the old boost amount.
-        h.maxHealth = Math.round(h.maxHealth / (1f + 0.01f*hBoost.boostAmount));
-
-        // If the current health is greater than the new max health, set the current health value to be the max health.
-        if (h.currentHealth > h.maxHealth) {
-            h.currentHealth = h.maxHealth;
-        }
+    private void removeBoost(EntityRef entity, HealthBoostComponent boost) {
+        entity.updateComponent(HealthComponent.class, health -> {
+            // Reverse the max health boosting effect by dividing the old boost amount.
+            health.maxHealth = Math.round(health.maxHealth / (1f + 0.01f * boost.boostAmount));
```

Comment:
Actually this won't change the healthBar UI, so the new maxHealth will be mapped to the original 10 icons..
I'd suggest using the new added ChangeMaxHealthEvent.. which will also take care of the health.currentHealth changes itself.
Same for all the places where maxHealth is modified in here..

Reply:
Good point, I'll update that! :+1: